### PR TITLE
Categories with extension filter

### PR DESCRIPTION
**Pull request by MStruller**
here are fixes for contentelement categories

### CODE REVIEW

File: admin/contentelements/translationExtensionFilter.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	}
+
+	public function createFilterHTML()
+	{
+		$db = JFactory::getDBO();
+
+		if (!$this->filterField)
+			return "";
+		$ExtensionOptions = array();
+		$ExtensionOptions[] = JHTML::_('select.option', $this->filterNullValue, JText::_('ALL_EXTENSIONS'));
```
Comment (MStruller) [inline admin/contentelements/translationExtensionFilter.php]: TODO add language string

File: admin/contentelements/translationExtensionFilter.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		$cats = $db->loadObjectList();
+		$catcount = 0;
+		foreach ($cats as $cat)
+		{
+			$ExtensionOptions[] = JHTML::_('select.option', $cat->extension, $cat->extension);
+			$catcount++;
+		}
+		$Extensionlist = array();
+		$Extensionlist["title"] = JText::_('EXTENSION_FILTER');
+		$Extensionlist["html"] = JHTML::_('select.genericlist', $ExtensionOptions, 'extension_filter_value', 'class="inputbox" size="1" onchange="document.adminForm.submit();"', 'value', 'text', $this->filter_value);
```
Comment (MStruller) [inline admin/contentelements/translationExtensionFilter.php:107]: TODO add com_content as default but where?

**Comment by quodo** [comment]
This is a nice addition to easily show the user which category belongs to what extension.

I like this filter becaus it solves this:
In the sample data there are five categories called "uncategorised", each with the same title (Uncategorised) and alias (uncategorised). In the table jos_categories you can tell by the extension field that these are for different extensions: com_content, com_banners, com_contact, com_newfeeds and com_weblinks. E.g, the "Category Manager: Banners" shows only its own Uncategorised category.
In JoomFish' translations screen for Content Element Categories you don't know which of which extension you're translating the category.

I can think of two improvements
- The list is now "com_content", "com_banners", etc., however, "articles", "banners" would be more usefull for an end user.
- I think the "system" extension should not be in the list. It is the root category (the only one wiht level 0) that is the parent for all other level 1 categories.
- (just for completeness: the language strings should be added).

(Tested it with JoomFish 2.5 alpha aaaf802a4 on which this pull request was applied as a patch. Patch file downloaded by adding .patch to the current URL: https://github.com/JoomFish/jf-future/pull/40.patch.)

File: admin/contentelements/translationExtensionFilter.php (hunk 2 above)
Comment (quodo) [inline admin/contentelements/translationExtensionFilter.php:107]: I prefer "ALL_EXTENSION" as default over "com_content" as default. 
Reason: now you see all categories and an end user will find the category (s)he's looking for, if the EXTENSION_FILTER needs to be changed first I image that people have more of a problem finding their category if it's not a com_content category.

**Comment by MStruller** [comment]
Hello klas

the "system" extension is already filter out in the xml with <filter>c.id > 1 </filter>

**Comment by MStruller** [comment]
i am thinking the <field type="text" name="extension" translate="0">extension</field> in  the xml not needed

**Comment by quodo** [comment]
> - I think the "system" extension should not be in the list. It is the root category (the only one wiht level 0) 
>   that is the parent for all other level 1 categories.

MStruller commented:

> the "system" extension is already filter out in the xml with <filter>c.id > 1 </filter>

To be more precise: the "system" extension should not be in the **dropdown** list **in the filter**.

**Comment by MStruller** [comment]
The system filter is not in dropdown only in the generall filter for the table
see https://github.com/MStruller/jf-future/commit/35c7f96f78edc80fc5fb472e05264f97fbcc595f line 17

**Comment by quodo** [comment]
I have that line 17 in /JOOMLAROOT/administrator/components/com_joomfish/contentelements/categories.xml:
`<filter>c.id > 1 </filter>`

This is what I do and see
- Backend > Components > JoomFish > Translation
- Set Languages to my 2nd language (dutch)
- Set Content Elements to Categories

Now your patch shows an extra filter (and I like it!). Currently it is called "EXTENSION_FILTER".

The list in that filter (in the dropdown) has for me
- ALL_EXTENSIONS
- com_content
- com_banners
- com_contact
- com_newsfeeds
- com_weblinks
- com_user.notes
- **system**

It is that system that I think should not be in the list to avoid confusion for the end-user. 

Are we talking about the same "system" entry? I think you are talking about a different "system". If I comment the mentioned line 17, 
`<filter>c.id > 1 </filter>`
then I get an extra item to translate. It's title is 'ROOT' and it is indeed the only item that is present if I use the EXTENSION_FILTER and choose 'system' (as opposed to for example com_banners). That 'ROOT' entry is indeed something I would not show to the user. However, it's a different entry than what I mean.

(I'd attach an image of what I mean if I knew how to add images to a comment, that'd much easier)

**Comment by MStruller** [comment]
i hav not the extension system in my dropdown
but i have rewrite the filter to c.level > 0

**Comment by MStruller** [comment]
Hello quodo

can you please check the **id** from entry system in the categories table and tell me?

**Comment by quodo** [comment]
Of course: From my jos_categories table (where jos is the prefix), Joomla 2.5.1 with sample data.

Record with id =1, asset_id = 0, parent_id = 0, level = 0, path = '', extension = **system**, title = ROOT, alias = root, description = '', published = 1 (and more).

On the JoomFish forum I've posted a new topic where an image is attached:
http://www.joomfish.net/forum/viewtopic.php?f=15&t=11164

Your new commits indeed remove the System from the dropdown. And even better, the component names are show, e.g. Articles instead of com_content.

Thanks! I hope this makes it into JoomFish :-)

**Comment by MStruller** [comment]
Hello Quodo
the same id for entry system but two results, why this?
can yo please try in the contentelements/translationExtensionFilter.php 

$query->where('id > 1'); old was // $query->where('id > \'1\'');
//$query->where('level > 0');

**Comment by quodo** [comment]
I think there were two results because you added some new commits yesterday afternoon. The image with "system" (and com_content, etc.) was made with your commits I took from Github yesterday afternoon. Today, there were some new commits, these fixed what I was talking about. You can see in the image with "articles" etc. in the dropdown.

I'd say it's ok now.

**Comment by MStruller** [comment]
Hello quodo
i mean the different between your and my working-system
so i want check is on your system work  with $query->where('id > 1'); instant from // $query->where('id > \'1\'');

**Comment by quodo** [comment]
With you latest commits I get the screenshot as in 
http://www.joomfish.net/forum/viewtopic.php?f=15&t=11164#p38054
file JF25alpha_translateCategories_extensionFilter_improved.jpg

This is when /JOOMLAROOT/administrator/components/com_joomfish/contenelements/translationExtensionFilter.php on line 73 is:
`//$query->where('id > \'1\'');`

You've asked 

> i want check is on your system work with 
> $query->where('id > 1'); instant from // $query->where('id > \'1\'');

so I uncommented that line 73 to
`$query->where('id > \'1\'');`

This does not change anything in the dropdown. The screen shows the same as in the mentioned JF25alpha_translateCategories_extensionFilter_improved.jpg.

_(I'll send you a PM via the joomfish website)_

**Comment by MStruller** [comment]
Hello quodo
not $query->where('id > \'1\'');
please
$query->where('id > 1');
unquotet remove the \' before and after 1

**Comment by quodo** [comment]
OK, changed the line 17 to 
$query->where('id > 1');
and nothing has changed, as expected (since there is also a line 
$query->where('level > 0');
and the only record for which level > 0 does not hold is the record with id 1).

At this point I do not understand what you want to achieve. To me, the dropdown already works OK. Please take a look at the screenshot JF25alpha_translateCategories_extensionFilter_improved.jpg in http://www.joomfish.net/forum./viewtopic.php?f=15&t=11164#p38054.

**Comment by MStruller** [comment]
Hello quodo
i will check if the \' make the different 
so also comment out $query->where('level > 0');
so  only $query->where('id > 1'); in the querry

**Comment by MStruller** [comment]
fix dropdown teyt must fix dropdown text

the two improvements where klas want are included and work
so it can merge if want this

**Pull request merged**